### PR TITLE
Add Archive Extraction Task

### PR DESCRIPTION
**Pull request by hacktobeer**
### Summary:
This PR introduces a new Celery task for extracting various archive formats (currently supporting 7zip) and integrates it into the workflow.

### Technical Details:
*   **New Extraction Task:**  A new Celery task, `extract_archive`, has been implemented in `src/extract_archive.py`. This task extracts files from supported archive formats.  It handles logging, output file creation, and associates extracted files with their original archive using the `source_file_id`.
*   **7zip Support:** The initial implementation utilizes the `p7zip-full` package for 7zip archive extraction. This package has been added to the Dockerfile.
*   **Celery Integration:** The new task has been registered with Celery in `src/app.py` to enable its execution within the existing workflow system.
*   **Output File Handling:** The `extract_archive` task creates output files for each extracted item, preserving the original file name and including the `original_path` within the archive and the `source_file_id` linking it back to the original archive.  This is crucial for maintaining context and provenance.
*   **Dependency Updates:** The `poetry.lock` file has been updated to reflect the addition of new dependencies and version changes. This ensures consistent dependency management across environments.

### CODE REVIEW

File: src/extract_archive.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        workflow_id: ID of the workflow.
+        task_config: User configuration for the task.
+
+    Returns:
+        Base64-encoded dictionary containing task results.
+    """
+    input_files = get_input_files(pipe_result, input_files or [])
+    output_files = []
+    for input_file in input_files:
+        print(f"Processing {input_file.get("path")}")
```
Comment (berggren) [inline src/extract_archive.py:63]:
```suggestion
```

File: src/extract_archive.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        log_file = create_output_file(
+            output_path,
+            filename=f"archive_extract_{input_file.get("display_name")}",
```
Comment (berggren) [inline src/extract_archive.py:66]:
```suggestion
            display_name=f"archive_extract_{input_file.get("display_name")}",
```

File: src/extract_archive.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        extracted_files = [f for f in export_directory_path.glob("**/*") if f.is_file()]
+        for file in extracted_files:
+            original_path = str(file.relative_to(export_directory_path))
+
+            output_file = create_output_file(
+                # TODO(rbdebeer) - fix when OutputFile class is refactored.
+                output_path=output_path,
+                filename=file.name,
+                original_path=original_path,
+                data_type="openrelik.worker.file.generic",
```
Comment (berggren) [inline src/extract_archive.py:86]:
```suggestion
```

File: src/extract_archive.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+        export_directory_path = Path(export_directory)
+        extracted_files = [f for f in export_directory_path.glob("**/*") if f.is_file()]
+        for file in extracted_files:
+            original_path = str(file.relative_to(export_directory_path))
+
+            output_file = create_output_file(
+                # TODO(rbdebeer) - fix when OutputFile class is refactored.
+                output_path=output_path,
+                filename=file.name,
```
Comment (berggren) [inline src/extract_archive.py:84]:
```suggestion
                display_name=file.name,
```

**Review by berggren — CHANGES_REQUESTED** [review]

**Comment by hacktobeer** [comment]
@berggren PTAL, small requested changes made.

File: src/extract_archive.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            output_files.append(log_file.to_dict())
+
+        export_directory_path = Path(export_directory)
+        extracted_files = [f for f in export_directory_path.glob("**/*") if f.is_file()]
+        for file in extracted_files:
+            original_path = str(file.relative_to(export_directory_path))
+
+            output_file = create_output_file(
+                # TODO(rbdebeer) - fix when OutputFile class is refactored.
+                output_path=output_path,
```
Comment (berggren) [inline src/extract_archive.py:82]:
```suggestion
                output_path,
```

File: src/extract_archive.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                source_file_id=input_file.get("uuid"),
+            )
+            os.rename(file.absolute(), output_file.path)
+
+            output_files.append(output_file.to_dict())
+
+        shutil.rmtree(export_directory)
+
+    if not output_files:
+        raise RuntimeError("archive extractor didn't create any output files")
```
Comment (berggren) [inline src/extract_archive.py:94]:
```suggestion
        raise RuntimeError("Archive extractor didn't create any output files")
```

**Review by berggren — APPROVED** [review]

**Pull request merged**